src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/ApplicationInsightsHostBuilderExtensions.cs

```diff
@@ -28,10 +30,11 @@ public static class ApplicationInsightsHostBuilderExtensions
         /// </summary>
         /// <param name="builder">The host builder.</param>
         /// <param name="instrumentationKey">The Application Insights instrumentation key.</param>
+        /// <param name="samplingSettings">The <see cref="SamplingPercentageEstimatorSettings"/> to use for configuring adaptive sampling. If null, sampling is disabled.</param>
         /// <returns>A <see cref="IHostBuilder"/> for chaining additional operations.</returns>
         public static IHostBuilder AddApplicationInsights(this IHostBuilder builder, string instrumentationKey, SamplingPercentageEstimatorSettings samplingSettings)
```

Comment:
This is a nit, and I realize this pattern was already here prior to this PR,  but can we have an overload without the `samplingSettings` argument (and ideally, validate that it isn't `null` in the overload that takes the parameter) since that is optional, and not just rely on `null` being passed? That will make it an explicit choice and a clearer API and avoid issues if a `null` reference is unintentionally passed. The

Reply:
Sure, I will do this. But I won't have time today, so please wait till tomorrow.

Thanks for the review!

Reply:
No need to block on this, as it could come later if there's any urgency on getting this in.

Thanks again for sending this!

Reply:
@fabiocav -- these settings are exposed so that Script can pass in settings via host.json. Should we try to move this to using `IOptions<>` so we can bind directly? Having (only) these settings exposed seems odd now.

Doesn't have to be done here, but we can file an issue to clean up in a later PR...

Reply:
Hey, I just created an issue for this: https://github.com/Azure/azure-webjobs-sdk/issues/1780.

So it seems `AddApplicationInsights` would eventually have IConfiguration/IOptions overload and some set of default parameters. 
For now, I believe having extra overloads that set default parameters is somewhat useless, so I'd rather remove it completely.

what do you think?